### ISSUE TITLE
Can't use variables inside Liquid Content property

### ISSUE THREAD

**Issue by MikeKry** [OrchardCMS/OrchardCore#10134]
After updating to newer OC version, my old shortcode
{% assign content_item = Content.Alias[Args.alias] %}
stopped working. 

I found in documentation, that it changed into 
{% assign content_item = Content["alias:some-alias"] %} 
but when I want to pass dynamic property into this liquid macro like this:
{% assign content_item = Content["alias:" + Args.alias] %} 

I get an error - The template doesn't contain a valid Liquid expression. Details: '%}' was expected at (1:33)

**Comment by jtkech** [comment]
Assign a var using the `| append: someString` and use it in place of `"alias:" + Args.alias`.

**Comment by Skrypt** [comment]
Yes, I second what @jtkech says. We can't use the + sign to make string concatenation anymore. It was not compliant with Liquid.

**Comment by MikeKry** [comment]
@jtkech 

I can not get it working.. 

![image](https://user-images.githubusercontent.com/52829889/129863169-1ba8e98b-d523-4adb-982f-a5dbf35a32a1.png)

it seems that it is not a string, when i use append like this.

**Comment by jtkech** [comment]
@MikeKry 

> {% assign alias_str = "alias:" | append: Args.alias %} is working, trumbowyg editor put tag into alias variable, so that is why it did not work.

Okay cool, yes feel free to suggest any PR that may improve the documentation.

**Comment by dodyg** [comment]
>Yes, I second what @jtkech says. We can't use the + sign to make string concatenation anymore. It was not compliant with Liquid.

Yeah I was bitten by this a couple of days ago lol.

**Comment by sebastienros** [comment]
Please file a PR that show how to create a variable like this, in the page you linked to. Thanks.

**Comment by davidhayden** [comment]
Not being able to concatenate using "+" has been removed for a bit now prior to the 1.0 release. It occurred without warning during a Fluid package update.

Unless something has changed recently, you also can't use "+" to do simple addition anymore, like `{% assign ctr = ctr + 1 %}`.

You have to use `{% assign ctr = ctr | plus:1 %}`.

I rather miss the magic of "+" in both cases as I think it's more intuitive and expressive.

**Comment by sebastienros** [comment]
I decided to strictly follow the standard, I am sorry that it broke things, but the conversion is usually obvious and that's necessary if we want better interoperability with other systems that use Liquid, for instance on the client-side (using LiquidJs). 
This also goes the other way around, take a look at this PR https://github.com/RicoSuter/NJsonSchema/pull/1346 that is trying to switch the liquid engine to use Fluid (for performance and compliance reasons). The previous library didn't follow the correct syntax (wrt space trimming), so now it's hard to convert to a different library.

The other advantage of `plus` vs `+` is that it's expressive in terms of type manipulation. What does `"0" + 1` mean?
Now `{{ "0" | plus: 1 }}` and `{{ "0" | append: 1 }}` don't have this issue.

And finally there is the risk that one day a change in the official Liquid makes the usage of standard operators incompatible with the new features.

PS: take a look a Scriban https://github.com/scriban/scriban which support liquid and lots of other extensions to the language, like functions. If we had known about it before I start developing Fluid I am pretty sure we would have used it. But we landed on Liquid, then tried DotLiquid but the perf was abysmal, so I decided to write a new library. We looked at many templating solution at that time.

**Comment by davidhayden** [comment]
I didn't mean to sound like I was complaining. Once I found `append` and `plus` in the liquid documentation I figured you were strictly following the standard, which is fine by me. 👍
